src/compiler/checker.ts

```diff
@@ -16498,7 +16498,7 @@ namespace ts {
             if (!links.resolvedType) {
                 // Deferred resolution of members is handled by resolveObjectTypeMembers
                 const aliasSymbol = getAliasSymbolForTypeNode(node);
-                if (getMembersOfSymbol(node.symbol).size === 0 && !aliasSymbol) {
+                if ((node.symbol === undefined || getMembersOfSymbol(node.symbol).size === 0) && !aliasSymbol) {

```

Comment:
How hardened should the checker be against type nodes with no symbol for the language service? Should some other part of the checker have ensured that the JSDoc type node had a symbol? @sandersn @jakebailey

Reply:
My uninformed understanding was that this should always be defined (hence, it not being potentially undefined in the Node type); is the binder not walking something?

Reply:
@jakebailey You are right, `binder` is not walking on `jsDoc` nodes in TypeScript files

https://github.com/microsoft/TypeScript/blob/08b91f6b827e9ee98407c0eb969579ab77fa9f0c/src/compiler/binder.ts#L2450-L2464

for this reason, we don't have a `symbol` in the TypeScript files for

https://github.com/microsoft/TypeScript/blob/08b91f6b827e9ee98407c0eb969579ab77fa9f0c/src/compiler/binder.ts#L2637-L2640

At the same time, we can trigger on/use that node in services (in TypeScript/JavaScript)
```ts
// a.ts
/**
 * @type {{ property: /**/"foo"}}
*/
```

in rename 
https://github.com/microsoft/TypeScript/blob/618fb2d8b92357d564ddb8596296ecaba5468002/src/services/rename.ts#L22-L23

or in find all 
https://github.com/microsoft/TypeScript/blob/6362fb2dceb9d4a6fc8640b3d82bca02961d142e/src/services/findAllReferences.ts#L2088-L2089

Reply:
That's what I thought. It feels like if this is the case, then we should have loads of issues, like crashing on hover of a jsdoc in a TS file, for example. But we don't, so does the services code defend against this and we're missing things?

(Also, can we add asserts to the checker to verify that we definitely always have a symbol when given a node?)

Reply:
My opinion is that 

1. `@type`, `@typedef` and other type-related jsdoc are meaningless in Typescript and shouldn't be maintained. 
2. That means that find-all-refs, rename, goto-def, et al should intentionally skip jsdoc in Typescript. As @jakebailey points out, we should have loads of crashes already so I think there *must* be some code to skip jsdoc in services already. I think the correct fix is to find that code and make it skip the examples from the bug too.

If there's not already code to skip this kind of jsdoc, and we've just been ignoring a whole segment of crashes, now is *definitely* the time to add it.